src/OSPSuite.Core/Domain/Mappers/MoleculeBuilderToMoleculeAmountMapper.cs

```diff
@@ -69,6 +73,7 @@ public MoleculeAmount MapFrom(MoleculeBuilder moleculeBuilder, IContainer target
             .WithDisplayUnit(_amountDimension.UnitOrDefault(_amountDimension.DefaultUnit.Name));
 
          simulationBuilder.AddBuilderReference(moleculeAmount, moleculeBuilder);
+         _objectTracker.TrackObject(moleculeAmount, moleculeBuilder, simulationBuilder);
```

Comment:
we already have the concept of "builder reference" so I am just adding on top of if for now. Can probably be removed when all done since we are tracking more